src/main/java/com/iexec/resultproxy/ipfs/IpfsService.java

```diff
@@ -62,4 +59,26 @@ public static boolean isIpfsHash(String hash) {
         }
     }
 
+    @Override
+    @Retryable(maxAttempts = 10)
+    public void start() {
+        this.ipfs = new IPFS(multiAddress);
+    }
+
+    @Recover
+    public void start(RuntimeException exception) {
+        log.error("Exception when initializing IPFS connection", exception);
+        log.warn("Shutting down the service since IPFS is necessary");
+        throw exception;
+    }
+
+    @Override
+    public void stop() {
+
```

Comment:
Very minor:
`log.warn("Shutting down service");` (and remove `log.warn("Shutting down the service since IPFS is necessary");` from `start` method)
?

Reply:
Ok for the first suggestion.
Sadly, the second won't work. As the service will not be effectively started, the `stop` method will not be called.